ui/src/app/pages/appAnalysis/app-selector.tsx

```diff
@@ -1,65 +1,63 @@
 import * as React from 'react';
 import { useState } from 'react';
-import { Dropdown } from 'primereact/dropdown';
 
 import { BillingStatus } from 'generated/fetch';
 
+import { UIAppType } from 'app/components/apps-panel/utils';
 import { Button } from 'app/components/buttons';
-import {
-  Modal,
-  ModalBody,
-  ModalFooter,
-  ModalTitle,
-} from 'app/components/modals';
 import { NewJupyterNotebookModal } from 'app/pages/analysis/new-jupyter-notebook-modal';
 import colors from 'app/styles/colors';
-import { reactStyles } from 'app/utils';
+import { reactStyles, switchCase } from 'app/utils';
 import { AnalyticsTracker } from 'app/utils/analytics';
-import { APP_LIST, JUPYTER_APP } from 'app/utils/constants';
+import { userAppsStore, useStore } from 'app/utils/stores';
+import { openRStudioOrConfigPanel } from 'app/utils/user-apps-utils';
 import { WorkspaceData } from 'app/utils/workspace-data';
 import { WorkspacePermissionsUtil } from 'app/utils/workspace-permissions';
 
+import { AppSelectorModal } from './app-selector-modal';
+
 const styles = reactStyles({
   startButton: {
     paddingLeft: '0.75rem',
     height: '3rem',
     backgroundColor: colors.secondary,
   },
-  appsLabel: {
-    color: colors.primary,
-    fontWeight: 600,
-    fontSize: '14px',
-    lineHeight: '24px',
-    paddingBottom: '0.75rem',
-  },
 });
 
+const enum VisibleModal {
+  None = 'None',
+  SelectAnApp = 'SelectAnApp',
+  Jupyter = 'Jupyter',
+}
+
 interface AppSelectorProps {
   workspace: WorkspaceData;
 }
 
 export const AppSelector = (props: AppSelectorProps) => {
   const { workspace } = props;
-  const [selectedApp, setSelectedApp] = useState('');
-  const [showSelectAppModal, setShowSelectAppModal] = useState(false);
-  const [showJupyterModal, setShowJupyterModal] = useState(false);
+  const { userApps } = useStore(userAppsStore);
+  const [selectedApp, setSelectedApp] = useState<UIAppType>(undefined);
+  const [visibleModal, setVisibleModal] = useState(VisibleModal.None);
 
   const canCreateApps =
     workspace.billingStatus === BillingStatus.ACTIVE &&
     WorkspacePermissionsUtil.canWrite(workspace.accessLevel);
 
   const onClose = () => {
-    setSelectedApp('');
-    setShowSelectAppModal(false);
-    setShowJupyterModal(false);
+    setSelectedApp(undefined);
+    setVisibleModal(VisibleModal.None);
   };
 
   const onNext = () => {
-    setShowSelectAppModal(false);
     switch (selectedApp) {
-      case JUPYTER_APP:
+      case UIAppType.JUPYTER:
         AnalyticsTracker.Notebooks.OpenCreateModal();
-        setShowJupyterModal(true);
+        setVisibleModal(VisibleModal.Jupyter);
+        break;
+      case UIAppType.RSTUDIO:
+        setVisibleModal(VisibleModal.None);
+        openRStudioOrConfigPanel(workspace.namespace, userApps);
```

Comment:
different behavior from Jupyter: the Jupyter modal lets you choose new file or existing file.

Opening a specific file on launch is not something we can do in RStudio, so this choice doesn't make sense.  Instead we go directly to the app, if possible.